src/app/shared/state/entry-wizard.service.ts

```diff
@@ -0,0 +1,177 @@
+    const registryEnum = this.convertSourceControlStringToEnum(repository.gitRegistry);
+    this.defaultService
+      .addWorkflow(registryEnum, repository.organization, repository.repositoryName)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (workflow: BioWorkflow) => {
+          this.alertService.detailedSuccess('Workflow ' + repository.gitRegistry + '/' + repository.path + ' has been added');
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+          this.updateRepoIsPresent(repository, false, false);
+        }
+      );
+  }
+
+  /**
+   * Removes a workflow for the given git repository
+   * @param repository git repository
+   */
+  removeWorkflowFromDatabase(repository: Repository, matSlideToggle: MatSlideToggle) {
+    this.updateRepoIsPresent(repository, false, false);
+    this.entryWizardStore.setLoading(true);
+    const registryEnum = this.convertSourceControlStringToEnum(repository.gitRegistry);
+    this.defaultService
+      .deleteWorkflow(registryEnum, repository.organization, repository.repositoryName)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (workflow: BioWorkflow) => {
+          this.alertService.detailedSuccess('Workflow ' + repository.gitRegistry + '/' + repository.path + ' has been deleted');
+          // move this to be called right away
+          // on error, will revert state back
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+          this.updateRepoIsPresent(repository, true, true);
+        }
+      );
+  }
+
+  /**
+   * Updates the given repository in the store with the new isPresent value
+   * @param repository repository to update
+   * @param isPresent whether or not workflow now exists in database
+   * @param canDelete whether or not workflow can now be deleted
+   */
+  updateRepoIsPresent(repository: Repository, isPresent: boolean, canDelete: boolean) {
+    const state = this.entryWizardQuery.getValue();
+    const gitRepos = state.gitRepositories;
+    const index = gitRepos.findIndex(repo => {
+      return repository.path === repo.path && repository.gitRegistry === repo.gitRegistry;
+    });
+
+    const newRepository: Repository = {
+      organization: repository.organization,
+      repositoryName: repository.repositoryName,
+      gitRegistry: repository.gitRegistry,
+      canDelete: canDelete,
+      present: isPresent,
+      path: repository.path
+    };
+
+    const updatedGitRepos = Object.assign([], gitRepos);
+    updatedGitRepos[index] = newRepository;
+    this.entryWizardStore.update({ gitRepositories: updatedGitRepos });
+  }
+
+  /**
+   * Converts a string source control to a Source Control enum
+   * @param sourceControl source control string
+   */
+  convertSourceControlStringToEnum(sourceControl: string): BioWorkflow.SourceControlEnum {
```

Comment:
Seems like there should be a simpler way of writing this, not familiar enough with Typescript to help though

Reply:
Tried looking but couldn't find a better approach that worked.